rust-runtime/smithy-http/src/result.rs

```diff
@@ -35,7 +35,10 @@ pub enum SdkError<E> {
     },
 
     /// An error response was received from the service
-    ServiceError { err: E, raw: operation::Response },
+    ServiceError {
+        err: E,
+        raw: Option<operation::Response>,
+    },
```

Comment:
There should probably be a separate SdkError type for Event Stream errors that takes a raw `Message`.